django/db/migrations/autodetector.py

```diff
@@ -563,6 +563,19 @@ def generate_created_models(self):
                 if isinstance(base, str) and "." in base:
                     base_app_label, base_name = base.split(".", 1)
                     dependencies.append((base_app_label, base_name, None, True))
+                    # Add dependencies for base model's removed fields that are added into newly
+                    # added sub models so that field removal operations for base models can take
+                    # place before new inherited models are created.
+                    prev_state_base_model = self.from_state.models.get((base_app_label, base_name), 0)
+                    curr_state_base_model = self.to_state.models.get((base_app_label, base_name), 0)
```

Comment:
I don't think `, 0` is needed. `get()` will return `None` in the case of no match, which is more appropriate, and still falsey for the below expression.

```suggestion
                    prev_state_base_model = self.from_state.models.get((base_app_label, base_name))
                    curr_state_base_model = self.to_state.models.get((base_app_label, base_name))
```